pkg/util/log/log.go

```diff
@@ -674,14 +715,24 @@ func WarnfStackDepth(depth int, format string, params ...interface{}) error {
 
 // WarncStackDepth logs at the warn level with context and the current stack depth plus the additional given one and returns an error containing the formated log message
 func WarncStackDepth(message string, depth int, context ...interface{}) error {
-	return logContextWithError(seelog.WarnLvl, func() { Warnc(message, context...) }, Logger.warn, message, false, depth, context...)
+	return logFormatContextWithError(seelog.WarnLvl, func() { Warnc(message, context...) }, Logger.warn, message, nil, false, depth, context...)
 }
 
 // Warnc logs at the warn level with context and returns an error containing the formated log message
 func Warnc(message string, context ...interface{}) error {
 	return WarncStackDepth(message, 1, context...)
 }
 
+// WarnfcStackDepth logs at the warn level with context and the current stack depth plus the additional given one and returns an error containing the formatted log message
+func WarnfcStackDepth(depth int, context []interface{}, message string, params ...interface{}) error {
```

Comment:
Why dp we need this function ? would the stack depth not always be 1 ?

Reply:
This function is used in https://github.com/DataDog/datadog-agent/blob/d3fb0fe3e70facbd8be94fca7359fb204cf1d368/pkg/collector/corechecks/checkbase.go#L167-L179 to display the correct source in log as CheckBase has a wrapper of `log.Warnc/Warnf`